crates/clarirs_core/src/algorithms/simplify.rs

```diff
@@ -493,22 +493,30 @@ impl<'c> Simplify<'c> for BitVecAst<'c> {
                 }
                 BitVecOp::Pow(arc, arc1) => {
                     simplify!(arc, arc1);
-
                     match (arc.op(), arc1.op()) {
                         (BitVecOp::BVV(base), BitVecOp::BVV(exp)) => {
-                            let exponent = exp.to_usize().unwrap_or(0); // Convert exponent to usize
+                            let exponent = exp.to_u64().unwrap_or(0);
```

Comment:
This is gonna be incorrect anywhere where exp can't fit into a u64.

Reply:
I implement the pow function in bitvec.rs and utilized BigUint for exponents.